packages/espressocash_app/lib/features/tokens/token_list.dart

```diff
@@ -1,58 +1,105 @@
-import 'package:collection/collection.dart';
-import 'package:json_annotation/json_annotation.dart';
-import 'package:meta/meta.dart' show visibleForTesting;
+import 'package:path/path.dart';
+import 'package:shared_preferences/shared_preferences.dart';
 
 import '../../config.dart';
+import '../../data/db/db.dart';
+import '../../di.dart';
+import '../../gen/assets.gen.dart';
+import 'data/token_repository.dart';
+import 'services/token_service.dart';
 import 'token.dart';
 
-part 'token_list.g.dart';
-
-// ignore: invalid_annotation_target, we need const here
-@JsonLiteral('solana.tokenlist.json', asConst: true)
-const Map<String, dynamic> _solanaTokenList = _$_solanaTokenListJsonLiteral;
-
-// ignore: invalid_annotation_target, we need const here
-@JsonLiteral('local.tokenlist.json', asConst: true)
-@visibleForTesting
-const Map<String, dynamic> localTokenList = _$localTokenListJsonLiteral;
-
 class TokenList {
   factory TokenList({
     int chainId = currentChainId,
-    @visibleForTesting Map<String, dynamic>? data,
+    TokenService? service,
   }) {
     if (_instance?.chainId == chainId) {
       // ignore: avoid-non-null-assertion, cannot be null here
       return _instance!;
     }
-    _instance = TokenList._(chainId: chainId, data: data ?? _solanaTokenList);
+    _instance = TokenList._(
+      chainId: chainId,
+      service: service ?? TokenService(_repository),
+    );
 
     // ignore: avoid-non-null-assertion, cannot be null here
     return _instance!;
   }
 
   TokenList._({
     required this.chainId,
-    required Map<String, dynamic> data,
-  }) : _parsedContent = ParsedContent.fromJson(data);
-
+    required this.service,
+  }) {
+    initialize();
+  }
   static TokenList? _instance;
 
-  final ParsedContent _parsedContent;
+  static final TokenListRepository _repository =
+      TokenListRepository(sl<MyDatabase>());
+
+  final TokenService service;
+
   final int chainId;
 
-  Iterable<Token> get _allTokens => _parsedContent.tokens;
+  Future<Token?> getTokenByMint(String mint) async {
```

Comment:
It never returns `null`.

Also, this method and `initialize` should rather be defined on repository, and all the clients should deal with repository directly.

Reply:
Fixed in [4c03680](https://github.com/espresso-cash/espresso-cash-public/pull/1498/commits/4c03680f81d8b86eb2dc88cfbdf1da12b7d2940d)